ktor-io/common/src/io/ktor/utils/io/Coroutines.kt

```diff
@@ -115,15 +64,23 @@ private fun <S : CoroutineScope> CoroutineScope.launchChannel(
     attachJob: Boolean,
     block: suspend S.() -> Unit
 ): ChannelJob {
+    val originJob = coroutineContext[Job]
     val job = launch(context) {
         if (attachJob) {
             channel.attachJob(coroutineContext[Job]!!)
```

Comment:
is it `channel.attachJob(originJob!!)` ?

Reply:
Thanks. Will fix